pages/1_orders_and_revenue.py

```diff
@@ -16,51 +20,330 @@
+            textposition='outside',
+            marker_color='#1f77b4',  # Blue color for revenue bars
+            textfont=dict(size=10)
+        ),
+        secondary_y=False,
+    )
+    
+    # Add orders line
+    fig.add_trace(
+        go.Scatter(
+            x=combined_data['month'],
+            y=combined_data['header_id'],
+            name='Orders',
+            mode='lines+markers+text',
+            text=combined_data['header_id'].astype(str),
+            textposition='top center',
+            textfont=dict(size=10, color='#ff7f0e'),
+            line=dict(color='#ff7f0e', width=3),  # Orange color for orders line
+            marker=dict(size=8)
+        ),
+        secondary_y=True,
+    )
+    
+    # Update layout
+    fig.update_layout(
+        legend=dict(x=0.01, y=0.99, bgcolor='rgba(255, 255, 255, 0.5)'),
+        height=500,  # Increase height for better visibility
+    )
+    
+    fig.update_xaxes(title_text="Month", tickformat='%b %Y')
+    fig.update_yaxes(
+        title_text="Revenue", 
+        secondary_y=False, 
+        tickprefix='$', 
+        tickformat='~s',
+        showgrid=False,  # Hide grid lines for Revenue axis
+        showline=False   # Hide axis line for Revenue axis
+    )
+    fig.update_yaxes(
+        title_text="Orders", 
+        secondary_y=True,
+        showgrid=False,  # Hide grid lines for Orders axis
+        showline=False   # Hide axis line for Orders axis
+    )
+    
+    # Adjust the y-axis range for Orders to reduce overlap
+    max_revenue = combined_data['total_amount'].max()
+    max_orders = combined_data['header_id'].max()
+    revenue_order_ratio = max_revenue / max_orders
+    
+    fig.update_yaxes(
+        range=[0, max_revenue * 1.2],  # Increase the range for Revenue axis
+        secondary_y=False
+    )
+    fig.update_yaxes(
+        range=[0, max_orders * (1 + (1 / revenue_order_ratio))],  # Adjust Orders axis proportionally
+        secondary_y=True
+    )
+    
+    # Ensure labels are above everything
+    fig.update_traces(textfont_size=10)
+    fig.update_traces(textposition='outside', selector=dict(type='bar'))
+    fig.update_traces(textposition='top center', selector=dict(type='scatter'))
+    
+    st.plotly_chart(fig, use_container_width=True)
 
-    with row1_col1:
-        st.markdown("**Total Revenue Over Time**")
+    # Product Revenue and New Customers charts 
+    col1, col2 = st.columns(2)
+    
+    with col1:
+        st.markdown("**Product By Revenue**")
+        product_revenue = data.groupby('product_name')['total_amount'].sum().reset_index()
+        product_revenue = product_revenue.sort_values(by='total_amount', ascending=True)
+        
+        # Create the figure manually
+        fig = go.Figure()
+        
+        for index, row in product_revenue.iterrows():
+            fig.add_trace(go.Bar(
+                y=[row['product_name']],
+                x=[row['total_amount']],
+                orientation='h',
```

Comment:
How do you feel the orientation of the chart looks when viewed alongside the other charts on the page? I feel like the vertical orientation looks a bit off and throws off the flow of the page. What are your thoughts? Should we consider flipping to horizontal for consistency?

Reply:
It's very minor but I associate a horizontal chart with time series. But let me try that and lmk what you think

Reply:
switched back to vertical